nexus/types/src/external_api/params.rs

```diff
@@ -75,6 +75,23 @@ pub struct SledSelector {
     pub sled: Uuid,
 }
 
+/// Parameters for `sled_set_provision_state`.
+#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema, PartialEq)]
+pub struct SledProvisionStateParams {
+    /// The provision state.
+    pub state: super::views::SledProvisionState,
+}
+
+/// Response to `sled_set_provision_state`.
+#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema, PartialEq)]
+pub struct SledProvisionStateResponse {
+    /// The old provision state.
+    pub old_state: super::views::SledProvisionState,
+
+    /// The new provision state.
+    pub new_state: super::views::SledProvisionState,
```

Comment:
See my prior comments -- we *can* support this return value, but it means we'll need to be more careful with the database transactional semantics.